### PR TITLE
fix: [CDS-68293]: upgrade fabric8 library to 6.5.1

### PR DESCRIPTION
**Pull request by abhinavsingh1196**
## Describe your changes
fabric8 library needs to be upgraded to work with snakeyaml 2.0, which was recently upgraded.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47301)

### CODE REVIEW

**Comment by abhinavsingh1196** [comment]
trigger uts

**Comment by abhinavsingh1196** [comment]
trigger feature-build

**Comment by abhinavsingh1196** [comment]
trigger feature-build

**Comment by abhinavsingh1196** [comment]
trigger uts

**Comment by tchaurasiya** [comment]
*[`400-rest/src/test/java/software/wings/sm/StateMachineExecutorTest.java` line 69 at r5](https://reviewable.io/reviews/harness/harness-core/47301#-NUf7WdgCc5Fufga0-0Q:-NUf7WdgCc5Fufga0-0R:b-94lxwh) ([raw file](https://github.com/harness/harness-core/blob/9b34bea7e5fb31c19f4c67903c359721c1a10ff7/400-rest/src/test/java/software/wings/sm/StateMachineExecutorTest.java#L69)):*
> ```Java
> import static org.apache.commons.lang3.RandomUtils.nextInt;
> import static org.assertj.core.api.Assertions.assertThat;
> import static org.junit.jupiter.api.Assertions.assertThrows;
> ```

is this removal related to fabric8, if yes we have other instances of this too.

**Comment by yogesh-chauhan** [comment]
*[`WORKSPACE` line 5528 at r5](https://reviewable.io/reviews/harness/harness-core/47301#-NUf8F0A9YGaVYkg1l59:-NUf8F0A9YGaVYkg1l5A:b-l2d77f) ([raw file](https://github.com/harness/harness-core/blob/f38648a5da047190abf23719196706e06e7d0a0f/WORKSPACE#L5528)):*
> ```Starlark
>         maven_test_artifact("io.dropwizard:dropwizard-testing:2.0.10"),
>         maven_test_artifact("io.fabric8:kubernetes-server-mock:6.5.1"),
>         maven_test_artifact("org.junit.jupiter:junit-jupiter-api:5.8.2"),
> ```

is this intentional ?

**Review by yogesh-chauhan — DISMISSED** [review]
> Reviewed 12 of 19 files at r1, 1 of 1 files at r2, 5 of 6 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47301)* status: 19 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team,k8s-team.

**Comment by tchaurasiya** [comment]
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 1094 at r5](https://reviewable.io/reviews/harness/harness-core/47301#-NUf8zy86uJ8J_4Eqwys:-NUf8zy86uJ8J_4Eqwyt:b9yt8tr) ([raw file](https://github.com/harness/harness-core/blob/f38648a5da047190abf23719196706e06e7d0a0f/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L1094)):*
> ```Java
>   private NonNamespaceOperation<ReplicationController, ReplicationControllerList,
>       RollableScalableResource<ReplicationController>>
>   rcOperations(KubernetesConfig kubernetesConfig, String namespace) {
> ```

do we also want to add kubernetesClient as try block resource?

**Comment by tchaurasiya** [comment]
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 1094 at r5](https://reviewable.io/reviews/harness/harness-core/47301#-NUf8zy86uJ8J_4Eqwys:-NUf9TatD1jjeu1uFIQL:b-h4n8yn) ([raw file](https://github.com/harness/harness-core/blob/f38648a5da047190abf23719196706e06e7d0a0f/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L1094)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

do we also want to add kubernetesClient as try block resource?
</blockquote></details>

Same goes for all such rc operations.

**Review by tchaurasiya — APPROVED** [review]
> Reviewed 12 of 19 files at r1, 1 of 1 files at r2, 5 of 6 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47301)* status: Waiting on team(s): del-team,k8s-team.

**Review by abhinavsingh1196 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47301)* status: Waiting on team(s): del-team,k8s-team. 

___
*[`WORKSPACE` line 5528 at r5](https://reviewable.io/reviews/harness/harness-core/47301#-NUf8F0A9YGaVYkg1l59:-NUfErTABRC67JHRAe5V:b-wmi9nx) ([raw file](https://github.com/harness/harness-core/blob/f38648a5da047190abf23719196706e06e7d0a0f/WORKSPACE#L5528)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

is this intentional ?
</blockquote></details>

Yes, it fails with bazel build error in 840-template service otherwise
`ERROR: 840-template-service/BUILD.bazel:105:13: no such target '@maven//:org_junit_jupiter_junit_jupiter_api': target 'org_junit_jupiter_junit_jupiter_api' not declared in package '' defined by 423ef43bb28d6d144a4f60a1d57f8899/external/maven/BUILD and referenced by '//840-template-service:tests'`
___
*[`400-rest/src/test/java/software/wings/sm/StateMachineExecutorTest.java` line 69 at r5](https://reviewable.io/reviews/harness/harness-core/47301#-NUf7WdgCc5Fufga0-0Q:-NUfFsPh9q_fdR-zv4UI:b-7ixrpj) ([raw file](https://github.com/harness/harness-core/blob/9b34bea7e5fb31c19f4c67903c359721c1a10ff7/400-rest/src/test/java/software/wings/sm/StateMachineExecutorTest.java#L69)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

is this removal related to fabric8, if yes we have other instances of this too.
</blockquote></details>

This change can be left as is, or I can revert it too. But there's no value add as such.
Tests should consistently use either jupiter-api or assertj. I've resolve the mixed usage here.
___
*[`960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java` line 1094 at r5](https://reviewable.io/reviews/harness/harness-core/47301#-NUf8zy86uJ8J_4Eqwys:-NUfCVkC0-xmKuoL4E1Z:brh3iz) ([raw file](https://github.com/harness/harness-core/blob/f38648a5da047190abf23719196706e06e7d0a0f/960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java#L1094)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Same goes for all such rc operations.
</blockquote></details>

discussed offline, all k8s v2 callers are sending a client to the below method which is closed properly.
v1 flows have been left as is.

**Comment by abhinavsingh1196** [comment]
trigger uts

**Review by abosii-harness — DISMISSED** [review]

**Comment by abhinavsingh1196** [comment]
trigger messagecheck

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 2 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47301)* status: 3 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by abosii-harness — APPROVED** [review]

**Comment by abhinavsingh1196** [comment]
trigger uts

File: 960-api-services/BUILD.bazel [hunk 1]
```diff
@@ -224,9 +224,12 @@ java_library(
         "@maven//:commons_io_commons_io",
         "@maven//:dev_morphia_morphia_core",
         "@maven//:io_fabric8_kubernetes_client",
+        "@maven//:io_fabric8_kubernetes_client_api",
```
Comment (raghuAtWings) [inline 960-api-services/BUILD.bazel:227]: we need additional libraries with this upgrade?

Comment (abhinavsingh1196) [reply 960-api-services/BUILD.bazel:227]: fabric8 moved some code from `kubernetes-client` module to the new `kubernetes-client-api` module during 5.x -> 6.x migration

**Review by raghuAtWings — APPROVED** [review]

**Pull request merged**